tests/python_package_test/test_engine.py

```diff
@@ -13,7 +13,7 @@
 import psutil
 import pytest
 from scipy.sparse import csr_matrix, isspmatrix_csc, isspmatrix_csr
-from sklearn.datasets import load_svmlight_file, make_blobs, make_multilabel_classification
+from sklearn.datasets import load_svmlight_file, make_blobs, make_multilabel_classification, make_regression
```

Comment:
instead of `make_regression()`, I think we should use the already-imported `utils.make_synthetic_regression()`.

https://github.com/microsoft/LightGBM/blob/c8712a9e083bfa0922f6e860f49a8a374816f43e/tests/python_package_test/utils.py#L121-L123

Since the test being introduced in this PR needs more than just `n_samples`, it's ok to change the signature of `utils.make_regression` to allow passing through more arguments. Like this:

```python
def make_synthetic_regression(n_samples=100, n_features=4, n_informative=2, random_state=42):
    return sklearn.datasets.make_regression(
        n_samples=n_samples, n_features=n_features, n_informative=n_informative, random_state=random_state
    )
```

Those functions in `utils` that use `@lru_cache` are preferred because if multiple tests call them with the same arguments, every call after the first one gets a cached result instead of re-computing the dataset. See the discussion in https://github.com/microsoft/LightGBM/pull/3486 for background.

Reply:
Done in Done in [33e3fd6](https://github.com/microsoft/LightGBM/pull/4827/commits/33e3fd6598de6be2a0b2ce4fc470627fca55db7f)